src/Widgets/ServiceRow.vala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2013-2018 elementary, Inc. (https://elementary.io)
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public
+ * License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ */
+
+public class OnlineAccounts.ServiceRow : Gtk.ListBoxRow {
+    public Ag.Provider provider { get; construct; }
+    public string description { get; construct; default = ""; }
+    public string title_text { get; construct set; default = ""; }
+
+    public ServiceRow (Ag.Provider provider) {
```

Comment:
ServiceRow is quite misleading with the libaccounts-glib object that it takes. It would probably be better to name it `ProviderRow`